app/client/src/components/GenerateMicroreactURL.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <div>
+    <div>
+      <button
+        v-if='!microreactToken'
+        @click='showModal()'
+        class='btn btn-block btn-standard btn-download'
+      >
+        Generate Microreact URL
+      </button>
+
+      <button
+        v-if='microreactToken && !results.perCluster[cluster]?.microreactURL'
+        @click='getMicroreactURL(cluster)'
+        class='btn btn-block btn-standard btn-download'
+      >
+        Generate Microreact URL
+      </button>
+
+      <a
+        v-if='microreactToken && results.perCluster[cluster]?.microreactURL'
+        :href='results.perCluster[cluster]?.microreactURL'
+        class='btn btn-block btn-standard btn-download'
+        target='_blank'
+        rel='noreferrer'
+      >
+        Visit Microreact URL
+      </a>
+    </div>
+
+    <Modal v-if='isModalVisible' @close='closeModal' class='modalFlex'>
+      <template v-if='tokenWasWrong' v-slot:header>
+        Your submitted token seems to be wrong
+      </template>
+      <template v-else v-slot:header> No Token submitted yet </template>
+
+      <template v-if='tokenWasWrong' v-slot:body>
+        <p>It seems that the Token you gave us was wrong.</p>
+        <p>
+          Please make sure you are using a correct Token.<br />
+          You can find your token in your
```

Comment:
Seems strange to capitalise token. Either way it's best to be consistent with the capitalisation.